### DISCUSSION TITLE
Blocked tab not displaying any blocked work items

### DISCUSSION THREAD

**Discussion by GarethDKing** [nbrown02/FlowViz#78] [Q&A] [answered]
The tab is empty in the report but within the project I have a good number of blocked work items.  This is indicated by the state 'blocked'.  Am I missing something simple?

**Comment by nbrown02** [comment, marked as answer]
Hi Gareth,

Thanks for reaching out. If you're using a state of blocked then this will be mapped to a column of blocked as well I assume?

This template won't calculate that for you if so, as blocked columns are [considered an anti-pattern](https://www.linkedin.com/pulse/blocked-bad-brett-maytom/). A kanban board reflects the steps of how value flows through your system, therefore having a state/column of blocked alludes to 'planning' to be blocked. 

The blocked field or a tag containing blocked should be used, with the item remaining in the column it has been blocked in. 

![image](https://user-images.githubusercontent.com/29369962/177196217-e404642d-59b1-4000-a2eb-5c48ba1e6664.png)

![image](https://user-images.githubusercontent.com/29369962/177196243-d56c5a24-6d92-48e6-b9ae-367979527ae2.png)

See also the [wiki page](https://github.com/nbrown02/FlowViz/wiki/4.-Blockers)

Thanks

**Reply by GarethDKing** [reply]
Thanks for this and for pointing me to the wiki.  I realise it is an anti-pattern but it is one we are currently in and want to deal with.  Any chance of allowing the 'blocked' state to be turned on / off for that tab?

**Reply by nbrown02** [reply]
Hey Gareth - it's not something I would be looking to do for the main template, as it does require quite a bit of work to do it for this as well as the other methods it currently accommodates (Tags/Blocked Field).

Best I could do is mimic this on a test project and change the current queries to account for blocking only via this way and share that with you - this would of course be a one off exercise :) However it shows you how you can do it...

**Reply by GarethDKing** [reply]
That would be extremely generous and kind of you.

**Reply by nbrown02** [reply]
Cool - will try take a look in the next day or so and see what I can do :)

**Reply by nbrown02** [reply]
I've setup a dummy project like so:

![image](https://user-images.githubusercontent.com/29369962/177627105-e294ccc6-0e0f-4f2d-ac78-c8fd18d13516.png)

Just need a few days so I have some decent dummy data to work off...

**Reply by nbrown02** [reply]
@GarethDKing download via [this link](https://1drv.ms/u/s!AjPC43sChgyKgmFYDY4zD7Oztaky?e=85fgV0), load data as per usual - will open on the blockers page and hopefully populate...

**Comment by GarethDKing** [comment]
Hi Nick,

Thank you so much for this!  Current blocked items shows 1 for number of days blocked for all items, all others look right, (but horrible from my point of view)!

Gareth

**Reply by nbrown02** [reply]
Ah - it could be set to count rather than sum I think, also worth checking that column is formatted as a number...if you're not sure how to do that I can update later and send the new version 😁

**Reply by GarethDKing** [reply]
Changed it and now looking very realistic.  Thank you so much for your help Nick!

**Reply by nbrown02** [reply]
No problem - feel free to give feedback on the AzDO marketplace in the reviews 😀

**Comment by GarethDKing** [comment]
Just did!
